addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -0,0 +1,250 @@
+                                    + msg.getRequestHeader().getURI().toString();
+                }
+                if (requestMap.get(DATA) != null) {
+                    msg.getRequestBody().setBody((String) requestMap.get(DATA));
+                    msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+                }
+                try {
+                    httpSender.sendAndReceive(msg);
+                } catch (IOException e) {
+                    progress.warn(
+                            Constant.messages.getString(
+                                    "automation.error.requestor.badnetwork",
+                                    this.getName(),
+                                    name,
+                                    e));
+                    return;
+                }
+                persistToHistoryAndSitesTree(msg);
+                Object o = requestMap.get(RESPONSECODE);
+                if (o instanceof Integer) {
+                    int expectedCode = (Integer) o;
+                    int receivedCode = msg.getResponseHeader().getStatusCode();
+                    if (receivedCode != expectedCode) {
+                        progress.warn(
+                                Constant.messages.getString(
+                                        "automation.error.requestor.codemismatch",
+                                        msg,
+                                        expectedCode,
+                                        receivedCode));
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public String getExtraConfigFileData() {
+        return "    requests:\n"
+                + "      - url:                          # URL of the request to be made\n"
+                + "        method:                       # A non-empty request method\n";
+    }
+
+    private void persistToHistoryAndSitesTree(HttpMessage msg) {
+        HistoryReference historyRef;
+        ExtensionHistory extHistory =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+        try {
+            historyRef =
+                    new HistoryReference(
+                            Model.getSingleton().getSession(), HistoryReference.TYPE_ZAP_USER, msg);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
+            return;
+        }
+
+        try {
+            ThreadUtils.invokeAndWait(
+                    () -> {
+                        extHistory.addHistory(historyRef);
+                        Model.getSingleton().getSession().getSiteTree().addPath(historyRef, msg);
+                    });
+        } catch (Exception e) {
+            LOG.error("Could not add message to sites tree.", e);
+        }
+    }
+
+    @Override
+    public String getType() {
+        return JOB_NAME;
+    }
+
+    @Override
+    public Order getOrder() {
+        return Order.EXPLORE;
```

Comment:
Sorry, just realised that this should be FIRST_EXPLORE so that it will appear before the spiders, otherwise it could appear after them which is almost certainly not what most people would want.